src/utils/filter-utils.js

```diff
@@ -369,6 +435,74 @@ export function filterData(dataset, filters) {
   return {data: filtered, filteredIndex, filteredIndexForDomain};
 }
 
+/**
+ * This returns the points to evaluate in polygon filter
+ * @param layer
+ * @param data
+ * @return {*[][]}
+ */
+export const getLayerPointsToFilter = (layer, data) => {
```

Comment:
each layer has a `.getPositionAccessor` method, here you can precalculate `getPosition` and pass data to it

```js
export const getPolygonFilterFunctor = (layer, filter) => {
  const getPosition = layer.getPositionAccessor();

  switch (layer.type) {
    case LAYER_TYPES.arc:
    case LAYER_TYPES.line:
      return data => {
        const pos = getPosition({data});
        return pos.every(Number.isFinite) && [
          [pos[0], pos[1]],
          [pos[3], pos[4]]
        ].every(point => isInPolygon(point, filter.value));
      }
    default:
      return data => {
        const pos = getPosition({data});
        return pos.every(Number.isFinite) && isInPolygon(pos, filter.value);
      }
  }
};
```